tools/generator-terraform/generator/mappings/assignment_direct.go

```diff
@@ -552,6 +556,41 @@ output.%[1]s = &%[3]s
 	return &line, nil
 }
 
+func (d directAssignmentLine) schemaToSdkMappingBetweenDictionaryOfSimpleTypes(mapping resourcemanager.FieldMappingDefinition, schemaField resourcemanager.TerraformSchemaFieldDefinition, sdkField resourcemanager.FieldDetails, _ *assignmentConstantDetails, _ string) (*string, error) {
+	if schemaField.ObjectDefinition.NestedObject == nil {
+		return nil, fmt.Errorf("the Schema Model %q Field %q was a Dictionary with no NestedObject", mapping.DirectAssignment.SchemaModelName, mapping.DirectAssignment.SchemaFieldPath)
+	}
+	if sdkField.ObjectDefinition.NestedItem == nil {
+		return nil, fmt.Errorf("the SDK Model %q Field %q was a Dictionary with no NestedItem", mapping.DirectAssignment.SdkModelName, mapping.DirectAssignment.SdkFieldPath)
+	}
+
+	v, ok := directAssignmentTypes[schemaField.ObjectDefinition.NestedObject.Type]
+	if !ok {
+		return nil, fmt.Errorf("a DirectAssignment wasn't defined between %q and %q", string(schemaField.ObjectDefinition.NestedObject.Type), string(sdkField.ObjectDefinition.NestedItem.Type))
+	}
+	if v != sdkField.ObjectDefinition.NestedItem.Type {
+		return nil, fmt.Errorf("expected a DirectAssignment between %q and %q but got %q", string(schemaField.ObjectDefinition.NestedObject.Type), string(v), string(sdkField.ObjectDefinition.NestedItem.Type))
+	}
+
+	if schemaField.Required {
+		line := fmt.Sprintf("output.%[1]s = input.%[2]s", mapping.DirectAssignment.SchemaFieldPath, mapping.DirectAssignment.SdkFieldPath)
+		if sdkField.Optional {
+			line = fmt.Sprintf("output.%[1]s = &input.%[2]s", mapping.DirectAssignment.SchemaFieldPath, mapping.DirectAssignment.SdkFieldPath)
```

Comment:
can we use `pointer.To` here:

```suggestion
			line = fmt.Sprintf("output.%[1]s = pointer.To(input.%[2]s)", mapping.DirectAssignment.SchemaFieldPath, mapping.DirectAssignment.SdkFieldPath)
```